### PR TITLE
Separate fingerprint and index generation

### LINKED ISSUE #19: Fingerprint is broken
It appears your fingerprint is being drawn from the indexing hash, meaning all elements inserted into a particular bucket will have the same hash.  It increases your false positive rate by 2^8, making this crate into an vanilla bloom filter with 1 hash function, butat that consumes 8 times the necessary space.  I'll try to send a pull request.

### PR DESCRIPTION
**Pull request by florianjacob**
Fixes #19.  Also document “FaI” a little.

### CODE REVIEW

File: src/util.rs [hunk 1]
```diff
@@ -1,49 +1,59 @@
...
     pub fp: Fingerprint,
     pub i1: usize,
     pub i2: usize,
 }
 
-fn get_hash<T: ?Sized + Hash, H: Hasher + Default>(data: &T) -> [u8; 4] {
-    let mut result = [0; 4];
+fn get_hash<T: ?Sized + Hash, H: Hasher + Default>(data: &T) -> (u32, u32) {
+    let result: u64;
+    // TODO: why the additional scope?
```
Comment (florianjacob) [inline src/util.rs]: @seiflotfy could you explain the reasoning for this additional scope inside the function to me?

Comment (seiflotfy) [reply src/util.rs]: I am confused myself why I did that... there is no need for it

**Comment by seiflotfy** [comment]
Dude this is great thanks for picking up... got me all excited to do more rust again :D

**Comment by florianjacob** [comment]
Glad to hear that! :D
I hope it's okay when I request your review, at least for non-minor changes? I don't want to take too much of your time, but it's quite useful to get your feedback. 😉

My roadmap:
* Release v0.3.2 now with fingerprinting fixed, and publish that to crates.io and thereby fix #18 and indirectly #10 via https://docs.rs - I'll see whether I can get that done today.
* Breaking change: Return a `Result` type from `add()` instead of panicking, which will also fix #11 
* Add a C API so rust-cuckoofilter can be embedded in non-rust programs and used via FFI. Maybe as different module in the same repo, rust-cuckoofilter-c or something like that.
* Release v0.4.0 due to the API change

@seiflotfy Is that okay with you? Is there something from your side you want to have in those releases?

File: src/util.rs (hunk 1 above)
Comment (florianjacob) [reply src/util.rs]: Ok, I thought maybe the Hasher needs to be dropped early… But I could not find something like that and my integration test works without it as well, so cleaned up in 824bced21e8ffc0eb6cdcdbf45a90068a3da3b42. ✨

**Comment by florianjacob** [comment]
0.3.2 is ready on github now, only needs a `cargo publish` from your side to get on crates.io and docs.rs. 😉

**Comment by seiflotfy** [comment]
On it

**Comment by seiflotfy** [comment]
Adding you to the Authors :D

**Comment by florianjacob** [comment]
Thanks! :)

**Pull request merged**